src/lib/generate.js

```diff
@@ -0,0 +1,34 @@
+import TemplateStyleSheet from './template-stylesheet.json'
+
+let GENERATED_FILE = null
+
+/**
+ * Reads the values from local storage
+ * Appends css attribute
+ * @returns array: data that can be converted to css file.
+ */
+const getData = () => [
+  ':root{\n',
+  '/* Generated using ShabadOS Theme Generator */\n',
+  ...Object.keys( TemplateStyleSheet )
+    .filter( key => localStorage[key] !== 'none' )
+    .filter( key => localStorage[key] !== 'inherit' )
+    .map( key => ( `${key}:${localStorage[key]}\n` ) ),
+  '}',
+]
+
+/**
+ * Converts the data[] into a blob for storage
+ * @returns blob URL
+ */
+const makeCssFile = () => {
+  const dataFile = new Blob( getData(), { type: 'text/plain' } )
+  if ( GENERATED_FILE !== null ) {
```

Comment:
hmm i think there's gonna be a better way of doing this that preserves function purity/keeps it inside the function